src/main/java/com/github/lordrex34/config/ConfigManager.java

```diff
@@ -57,11 +49,11 @@
 {
 	private static final Logger LOGGER = LoggerFactory.getLogger(ConfigManager.class);
 	
-	/** A simple {@link AtomicBoolean} that indicates reloading process. */
-	private final AtomicBoolean _reloading = new AtomicBoolean(false);
+	/** Contains all the registered {@link ConfigClassInfo}s. */
+	private final Set<ConfigClassInfo> _configRegistry = new HashSet<>();
```

Comment:
I don't see an equals+hashCode on the ConfigClassInfo

Reply:
Done.